src/Terrain/Location.java

```diff
@@ -50,9 +50,14 @@ public void setOwner(Commander owner)
     }
   }
 
-  public Unit getResident()
+  public Unit getResident(GameMap map)
```

Comment:
I don't like needing to pass the map in here. I might look into alternatives.

Reply:
I did. They're all worse.
Unless you have a better?